Description: Cherrypick CMake changes from 4.7.4 to fix crashes and pixmap loadings.
 TODO: Put a short summary on the line above and replace this paragraph
 with a longer explanation of this change. Complete the meta-information
 with other relevant fields (see below for details). To make it easier, the
 information below has been extracted from the changelog. Adjust it or drop
 it.
 .
 veyon (4.7.3+repack1-1) unstable; urgency=medium
 .
   * New upstream release.
   * debian/copyright:
     + Update auto-generated copyright.in file.
     + Update copyright attributions.
   * debian/docs:
     + Drop CONTRIBUTORS. File vanished upstream.
   * debian/patches:
     + Trivial rebase of 2001_disable-CPackDefinitions-cmake.patch.
     + Drop 2002_no-bundled-kitemmodels.patch. Feature provided upstream.
     + Add 1001_fix-external-x11vnc.patch. Fix build with external x11vnc.
   * debian/veyon-plugins.install:
     + Simplify file, make it more robust to plugin changes.
   * debian/rules:
     + Obtain upstream tarball from Git releases (and import Git submodules
       when creating it).
     + Build with -DWITH_UNITY_BUILD=OFF.
   * debian/watch:
     + Fix for URL changes on Github.
   * debian/control:
     + Adapt S: fields as proposed by upstream.
     + Bump Standards-Version: to 4.6.0. No changes needed.
Author: Mike Gabriel <sunweaver@debian.org>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: 2022-08-22

--- veyon-4.7.3+repack1.orig/CMakeLists.txt
+++ veyon-4.7.3+repack1/CMakeLists.txt
@@ -46,7 +46,11 @@ if(COMMAND CMAKE_POLICY)
 		include(CheckPIESupported)
 		check_pie_supported(LANGUAGES CXX)
 		if(CMAKE_CXX_LINK_PIE_SUPPORTED)
-			set(CMAKE_COMPILE_FLAG_PIE "-fPIE")
+		set(CMAKE_COMPILE_OPTIONS_PIE "-fPIE")
+			set(CMAKE_LINK_OPTIONS_PIE "-pie;-fPIE;-fPIC")
+		else()
+			set(CMAKE_COMPILE_OPTIONS_PIE "-fPIC")
+			set(CMAKE_LINK_OPTIONS_PIE "-fPIC")
 		endif()
 	endif()
 endif()
@@ -271,7 +275,7 @@ set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 
-#include(cmake/CPackDefinitions.cmake)
+include(cmake/CPackDefinitions.cmake)
 
 
 # make sub-directories
--- veyon-4.7.3+repack1.orig/cmake/modules/BuildVeyonApplication.cmake
+++ veyon-4.7.3+repack1/cmake/modules/BuildVeyonApplication.cmake
@@ -11,8 +11,9 @@ macro(build_veyon_application APPLICATIO
 		install(TARGETS ${APPLICATION_NAME} RUNTIME DESTINATION bin)
 	endif()
 	target_include_directories(${APPLICATION_NAME} PRIVATE ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/src)
+        set_target_properties(${APPLICATION_NAME} PROPERTIES COMPILE_OPTIONS "${CMAKE_COMPILE_OPTIONS_PIE}")
+	set_target_properties(${APPLICATION_NAME} PROPERTIES LINK_OPTIONS "${CMAKE_LINK_OPTIONS_PIE}")
 	target_link_libraries(${APPLICATION_NAME} PRIVATE veyon-core)
-	set_property(TARGET ${APPLICATION_NAME} PROPERTY POSITION_INDEPENDENT_CODE TRUE)
 	set_default_target_properties(${APPLICATION_NAME})
 	if(WITH_PCH)
 		target_precompile_headers(${APPLICATION_NAME} REUSE_FROM veyon-application-pch)
--- veyon-4.7.3+repack1.orig/core/CMakeLists.txt
+++ veyon-4.7.3+repack1/core/CMakeLists.txt
@@ -111,7 +111,8 @@ if(WITH_PCH)
 	target_precompile_headers(veyon-core PRIVATE ${PCH})
 
 	add_pch_target(veyon-application-pch ${PCH})
-	set_target_properties(veyon-application-pch PROPERTIES COMPILE_FLAGS "${CMAKE_COMPILE_FLAG_PIE}")
+        set_target_properties(veyon-application-pch PROPERTIES COMPILE_OPTIONS "${CMAKE_COMPILE_OPTIONS_PIE}")
+	set_target_properties(veyon-application-pch PROPERTIES LINK_OPTIONS "${CMAKE_LINK_OPTIONS_PIE}")
 
 	add_pch_target(veyon-library-pch ${PCH})
 	set_target_properties(veyon-library-pch PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
